.github/workflows/gem-check.yml

```
@@ -0,0 +1,19 @@
+name: Check if the gem version exists on rubygems.org
+
+on: [pull_request]
+
+jobs:
+  build-gem:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Compare local and remote gem versions
+        run: |
+          LOCAL_GEM_VERSION=$(cat lib/scc/codestyle/version.rb | grep VERSION | awk -F"'" '{print $2}')
+          REMOTE_GEM_VERSION=$(curl 'https://rubygems.org/api/v1/versions/scc-codestyle/latest.json' | jq '.version' | tr -d \");
+          if [ "$(printf '%s\n' "$LOCAL_GEM_VERSION" "$REMOTE_GEM_VERSION" | sort -rV | head -n 1)" != "$REMOTE_GEM_VERSION" ]; then (echo "All good!" && echo exit 0); else (echo "Bump the gem version!" && exit 1); fi
```

~~small~~ big suggestion

```suggestion
          LOCAL_GEM_VERSION=$(ruby -I lib -e "require 'scc/codestyle'; print Scc::Codestyle::VERSION")
          REMOTE_GEM_VERSION=$(curl 'https://rubygems.org/api/v1/versions/scc-codestyle/latest.json' | jq -r '.version')
          LATEST_VERSION=$(printf '%s\n' "$LOCAL_GEM_VERSION" "$REMOTE_GEM_VERSION" | sort -rV | head -n 1)

          if [ "$LATEST_VERSION" == "$REMOTE_GEM_VERSION" ]; then
            echo "Bump the gem version!"
            exit 1            
          fi
```

using ruby to get the version also does syntax check as a bonus hehe, also less text processing and you can use full bash syntax (with spaces & indents) since the block is defined with `|` .

e.g.: `run: |` 

Besides that: I liked the latest version detection, pretty neat!